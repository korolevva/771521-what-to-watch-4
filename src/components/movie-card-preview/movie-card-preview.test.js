import React from "react";
import renderer from "react-test-renderer";
import MovieCardPreview from "./movie-card-preview.jsx";

const onCardTitleClick = (event) => {
  event.preventDefault();
};

const card = {
  src: `img/bohemian-rhapsody.jpg`,
  title: `Bohemian Rhapsody`,
};

it(`Render MovieCardPreview`, () => {
  const tree = renderer
    .create(<MovieCardPreview
      name={card.src}
      link={card.title}
      card={card}
      onCardTitleClick={onCardTitleClick}
      onCardHover={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
