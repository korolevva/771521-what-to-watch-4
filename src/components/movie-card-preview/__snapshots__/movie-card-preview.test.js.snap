// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCardPreview 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="img/bohemian-rhapsody.jpg"
      height="175"
      src="Bohemian Rhapsody"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      img/bohemian-rhapsody.jpg
    </a>
  </h3>
</article>
`;
